lib/inc/internal/detectors/virtualbox_detector.hpp

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <internal/sources/dmi_source.hpp>
+
+namespace whereami { namespace detectors {
+
+    /**
+     * VirtualBox detector function
+     * @param dmi_source An instance of a DMI data source
+     * @return Whether this machine is a VirtualBox guest
```

Comment:
Doxygen will automatically populate the type information in the docs (including cross-references) from the function signatures. One of the joys of a typed language :)